service-base/src/main/java/org/eclipse/hono/service/management/tenant/EventBusTenantManagementAdapter.java

```diff
@@ -201,38 +201,19 @@ protected final String getEventBusAddress() {
      * @return boolean The result of the check : {@link Boolean#TRUE} if the payload is valid, {@link Boolean#FALSE} otherwise.
      * @throws NullPointerException If the payload is {@code null}.
      */
-    private boolean isValidRequestPayload(final JsonObject payload) {
+    boolean isValidRequestPayload(final JsonObject payload) {
 
-        return hasValidAdapterSpec(payload) && hasValidTrustedCaSpec(payload);
-    }
+        Objects.requireNonNull(payload);
 
-    /**
-     * Checks if a payload contains a valid protocol adapter specification.
-     *
-     * @param payload The payload to check.
-     * @return boolean {@code true} if the payload is valid.
-     */
-    private boolean hasValidAdapterSpec(final JsonObject payload) {
-
-        final Object adaptersObj = payload.getValue(RegistryManagementConstants.FIELD_ADAPTERS);
-        if (adaptersObj instanceof JsonArray) {
-
-            final JsonArray adapters = (JsonArray) adaptersObj;
-            if (adapters.size() == 0) {
-                // if given, adapters config array must not be empty
-                return false;
-            } else {
-                final boolean containsInvalidAdapter = adapters.stream()
-                        .anyMatch(obj -> !(obj instanceof JsonObject) ||
-                                !((JsonObject) obj).containsKey(RegistryManagementConstants.FIELD_ADAPTERS_TYPE));
-                if (containsInvalidAdapter) {
-                    return false;
-                }
+        try {
+            payload.mapTo(Tenant.class);
```

Comment:
this is not equivalent to the original checks because the *setAdapters* method doesn't prevent setting an empty list (yet).

Reply:
@sophokles73 I have added a separate method for the above validation and pushed the changes. Pls take a look.